shell/common/gin_converters/image_converter.cc

```diff
@@ -27,9 +27,19 @@ bool Converter<gfx::Image>::FromV8(v8::Isolate* isolate,
   if (val->IsNull())
     return true;
 
+  // First see if the user has passed a path.
   gin::Handle<electron::api::NativeImage> native_image;
-  if (!gin::ConvertFromV8(isolate, val, &native_image))
-    return false;
+  base::FilePath icon_path;
+  if (gin::ConvertFromV8(isolate, val, &icon_path)) {
+    native_image =
+        electron::api::NativeImage::CreateFromPath(isolate, icon_path);
+    if (native_image->image().IsEmpty())
+      return false;
+  } else {
```

Comment:
this is used in `app.dock.setIcon` which takes a nativeImage or path - this would be the only place this gets handled

Reply:
possibly would be better to change that method to take `v8::Local<v8::Value>` instead of `gfx::Image` and use the new `TryConvertNativeImage`? That way you'd get better (+ consistent) error messages from `app.dock.setIcon`